src/shared/install.c

```diff
@@ -1755,6 +1757,10 @@ int unit_file_mask(
                 _cleanup_free_ char *path = NULL;
                 int q;
 
+                r = install_info_discover(scope, &c, &paths, *i, SEARCH_LOAD, &j);
+                if (r < 0)
+                        return r;
```

Comment:
Probably it's a better it would be better to do a dbus `GetAll` call and look for load state 'not-found' and active state inactive before we call this method.